packages/swap/src/translators/reactjs.tsx

```diff
@@ -131,6 +131,29 @@ export const useOrderByStatusOpen = (
   return query.data
 }
 
+export const usePairListByToken = (
+  tokenIdBase: Balance.Token['info']['id'],
+  options: UseQueryOptions<
+    Balance.Token[],
+    Error,
+    Balance.Token[],
+    ['persist:', 'usePairListByToken', string]
+  >,
+) => {
+  const {pairs} = useSwap()
+  const query = useQuery({
+    suspense: true,
+    ...options,
+    queryKey: ['persist:', 'usePairListByToken', tokenIdBase],
+    queryFn: () => pairs.list.byToken(tokenIdBase),
+  })
```

Comment:
@stackchain not sure about the `tokenIdBase` , from what I see in the
` byToken: getTokens,`  . The `getTokens` function in the openswap is receiving `{policyId = '', assetName = ''} ` not a string but maybe I understand it wrong